ntbs-build/openshift-pipelines/pipelines/ntbs-publish-database.yml

```diff
@@ -0,0 +1,202 @@
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+  name: ntbs-publish-database
+  namespace: ntbs-build
+spec:
+  params:
+    - default: 'yes'
+      description: Do you want to deploy the migration database?
+      name: deploy-migration
+      type: string
+    - default: 'yes'
+      description: Do you want to deploy the specimen-matching database?
+      name: deploy-specimen-matching
+      type: string
+    - default: 'yes'
+      description: Do you want to deploy the reporting database?
+      name: deploy-reporting
+      type: string
+    - default: uat
+      description: The environment you are deploying to (uat or live)
+      name: environment
+      type: string
+    - default: live
+      description: The branch to deploy
+      name: branch
+      type: string
+  tasks:
+    - name: git-clone-migration
+      params:
+        - name: url
+          value: 'git@github.com:publichealthengland/ntbs-data-migration.git'
+        - name: revision
+          value: $(params.branch)
+        - name: subdirectory
+          value: data-migration
+      taskRef:
+        kind: Task
+        name: git-clone-with-ssh
+      when:
+        - Input: $(params.deploy-migration)
+          Operator: in
+          Values:
+            - 'yes'
+      workspaces:
+        - name: output
+          workspace: shared-workspace
+    - name: download-migration
+      params:
+        - name: database
+          value: migration
+        - name: container
+          value: data-migration
+        - name: commit_hash
+          value: $(tasks.git-clone-migration.results.commit)
+      runAfter:
+        - git-clone-migration
+      taskRef:
+        kind: Task
+        name: download-dacpac
+      workspaces:
+        - name: output
+          workspace: shared-workspace
+    - name: publish-migration
+      params:
+        - name: checkout_directory
+          value: data-migration
+        - name: database
+          value: migration
+        - name: config_map
+          value: $(params.environment)
+        - name: publish_profile_key
+          value: databases.migration-publish-profile
+        - name: reference_dacpacs_path
+          value: ntbs-data-migration/data
+        - name: sql-user-secret
+          value: $(params.environment)-sql-keytab
```

Comment:
Currently I'm using this name based convention to tie the keytab secret to the environment. (The environment is also used to select the publish profile paths via the config map). I'm not thrilled about this, but the alternative would have been to add a separate input, which would have been annoying for people running the pipeline.